pipelines/pro_image.go

```diff
@@ -0,0 +1,87 @@
+package pipelines
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"dagger.io/dagger"
+	"github.com/grafana/grafana-build/containers"
+)
+
+func ProImage(ctx context.Context, dc *dagger.Client, directory *dagger.Directory, args PipelineArgs) error {
+	if len(args.PackageInputOpts.Packages) > 1 {
+		return fmt.Errorf("only one package is allowed: packages=%+v", args.PackageInputOpts.Packages)
+	}
+	packages, err := containers.GetPackages(ctx, dc, args.PackageInputOpts, args.GCPOpts)
+	if err != nil {
+		return fmt.Errorf("getting packages: packages=%+v %w", args.PackageInputOpts.Packages, err)
+	}
+
+	debianPackageFile := packages[0]
+
+	hostedGrafanaRepo, err := containers.CloneWithGitHubToken(dc, args.ProImageOpts.GithubToken, "https://github.com/grafana/hosted-grafana.git", "main")
```

Comment:
I would use `args.GrafanaOpts.GithubToken` here because it will be automatically populated by the environment if it's not set, like if you have `gh` set up it will use that.

Reply:
I'm seeing the token get populated in `GrafanaOpts.Grafana` and `GrafanaOpts.Enterprise` but not elsewhere 🤔 Did you mean to say that we should add a method to clone the hosted grafana repo to `GrafanaOpts`?

Reply:
oh you're right, it gets populated at that point in time.

I don't think it's super important to reimplement that we could probably do it a better way later.